akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -668,6 +668,8 @@ final case class Range(rangeUnit: RangeUnit, ranges: immutable.Seq[ByteRange]) e
 }
 
 final case class RawHeader(name: String, value: String) extends jm.headers.RawHeader {
+  require(name != null, "name must not be null")
```

Comment:
@ktoso I remember your comment about `require` allocating a closure in another PR. So I guess we should use here as well `if(name == null) throw ...`.

Reply:
True here as well, please if-throw instead.

Reply:
Ah good catch @jlprat! thanks! :)

Reply:
It got me by surprise on the other PR, so I tried to remember this one :)

Reply:
There's quite a few similar `require` calls in the file. Should those be changed too?

Reply:
Yeah, good idea